sympy/core/tests/test_evalf.py

```diff
@@ -535,3 +536,12 @@ def test_issue_14601():
     e2 = e.evalf(subs=subst)
     assert float(e2) == 0.0
     assert float((x + x*(x**2 + x)).evalf(subs={x: 0.0})) == 0.0
+
+
+def test_issue_11151():
+    e=Sum(0,(x,1,2))
+    array1=[]
+    array2=[]
+    array1.append(evalf.evalf(e, 15, {}))
+    array2.append(evalf.evalf(S(0), 15, {}))
+    assert array1 == array2
```

Comment:
This test does not show that there will be no error. Please add a test relevant to the issue.

Reply:
Ok I am adding simplify(expr4) == 0

Reply:
Yes there is no error because I have changed mpf(0) to None and prec value to prec

Reply:
A more direct test is given in the [issue here](https://github.com/sympy/sympy/issues/11151#issuecomment-473654255):
```python
z = S.Zero
e = Sum(z, (a, 1, 2))
assert e != z  # it shouldn't evaluate
# when it does evaluate, this is what it should give
assert evalf.evalf(e, 15, {}) == evalf.evalf(z, 15, {}) == (None, None, 15, None)
```

Reply:
Both of the last two assertions fail in master, the last because of the `TypeError: 'mpf' object is not iterable`

Reply:
Shall I add all three test cases for #11151 ?

Reply:
I have added

Reply:
I think that the test in [this comment](https://github.com/sympy/sympy/issues/11151#issuecomment-473603000) should be included. Otherwise, there will be no obvious connection with the issue.

Reply:
I have added it also

Reply:
> there will be no obvious connection with the issue

Once the root cause has been identified, I don't think it's necessary to add the more complex manifestation. But opinions vary on this.

Reply:
If you look at the traceback provided by @oscarbenjamin there is a line where the evaluation takes place and this precipitates the error. The fact that simplify was involved is secondary.
```
~/current/sympy/sympy/sympy/core/expr.py in _eval_is_positive(self)
    794                 return None
    795 
--> 796             n, i = self.evalf(2).as_real_imag()
    797             if not i.is_Number or not n.is_Number:
    798                 return False
```

Reply:
Yes.I couldn't understand what you want to say in the above example.

Reply:
The root issue is that the wrong value is being passed back. And this test asserts that it has been fixed:
```
z = S.Zero
e = Sum(z, (a, 1, 2))
assert e != z  # it shouldn't evaluate
# when it does evaluate, this is what it should give
assert evalf.evalf(e, 15, {}) == evalf.evalf(z, 15, {}) == (None, None, 15, None)
```
The error was noticed when this expression was being evaluated:
```
expr0 = Sum(a**2+a,(a,1,2))
expr1 = Sum(0,(a,1,2))
expr2 = expr1/expr0
expr3 = factor(expr2)
expr4 = expr3-expr2
expr5 = simplify(expr4)
```
But the only reason that it appeared there was because evalf was being done. And a simpler expression that would cause the same issue is
```
(Sum(0, (i, 1, 2))/2).n()
```
Testing manifestations feels good because you know that *that* error will never be allowed again. This is the "no regression" policy. 

What is wrong, however, is that there is nothing in those particular cases that guarantees that the issue is resolved. One could modify `simplify` so it detects a trivial sum and doesn't even give it to evalf. Then the problem would go away even though the root cause remained. For example, if the `simplify` path for this expression included `factor_terms(expr4)` this would give 0 and the evalf line might not be reached.

You've identified the root cause and tested it. You've also included the original issue which gives some context to the problem.

Reply:
Ok Let me try this on my system. This seems fine to me.